### PR TITLE
(SIMP-6307) Add custom audit profile

### PR DESCRIPTION
**Pull request by trevor-vaughan**
* Added a `custom` audit profile that accepts either an Array of rules
  or a template path for ease of setting full rulesets via Hiera.
* Updated all module components for `puppet strings`
* Fixed the README
* Added a REFERENCE.md
* Refactored the filename logic in the base profiles to be simpler
* Converted the rule template to EPP
* Converted the rotated_audit_logs template to EPP
* Converted STIG audit profile template to EPP
* Converted SIMP audit profile template to EPP

SIMP-6307 #close

### CODE REVIEW

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-auditd/pipelines/58504927

File: manifests/init.pp [hunk 1]
```diff
@@ -167,7 +170,7 @@
   Auditd::MaxLogFileAction                $max_log_file_action     = 'ROTATE', # CCE-27237-7
   Integer[0]                              $admin_space_left        = 50,
   Auditd::SpaceLeftAction                 $admin_space_left_action = 'SUSPEND', # CCE-27239-3 : No guarantee of e-mail server so sending to syslog.
-  Integer[0]                              $space_left              = $admin_space_left + 25, # needs to be larger than $admin_space_left or auditd will not start
+  Integer[0]                              $space_left              = $admin_space_left + 25, # needs to be larger than     $admin_space_left or auditd will not start
```
Comment (lnemsick-simp) [inline manifests/init.pp]: why did you add space in the comment?

File: manifests/rule.pp [hunk 2]
```diff
@@ -1,55 +1,68 @@
...
+#
+# @author https://github.com/simp/pupmod-simp-auditd/graphs/contributors
 #
 define auditd::rule (
-  String  $content,
-  Boolean $first    = false,
-  Boolean $absolute = false,
-  Boolean $prepend  = false
+  Variant[Array[String[1]],String[1]] $content,
+  String[1]                           $order    = '10',
```
Comment (lnemsick-simp) [inline manifests/rule.pp]: This new parameter, `$order`, is unused.

File: spec/acceptance/suites/compliance/00_simp_profile_install_spec.rb [hunk 3]
```diff
@@ -48,6 +48,10 @@
         host.reboot
       end
 
+      it 'will apply to cleanup' do
```
Comment (lnemsick-simp) [inline spec/acceptance/suites/compliance/00_simp_profile_install_spec.rb]: why was this extra puppet run required in order to converge?

File: spec/acceptance/suites/compliance/20_stig_profile_install_spec.rb [hunk 4]
```diff
@@ -58,6 +58,10 @@
         host.reboot
       end
 
+      it 'will apply to cleanup' do
```
Comment (lnemsick-simp) [inline spec/acceptance/suites/compliance/20_stig_profile_install_spec.rb]: Extra run?

File: templates/rule.epp [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<%- |
+  Variant[Array[String[1]], String[1]] $content
+|-%>
+<%
+  if $content =~ Array {
+-%>
+<%= join(strip($content).filter|$x| { !empty($x) }, "\n") %>
```
Comment (lnemsick-simp) [inline templates/rule.epp]: I didn't see a test for this for auditd::rule (Array input) nor did I see a use of auditd::rule with Array content throughout the module.  Please add a test.

File: manifests/config/audit_profiles.pp [hunk 6]
```diff
@@ -9,25 +9,31 @@
 # - `00_head.rules`:  Contains `auditctl` general configuration to
 #   remove existing rules when the rules are reloaded, ignore rule
 #   load errors/failures, and set the buffer size, failure mode,
-#   and rate limiting.
+#   and rate limiting
 # - `05_default_drop.rules`: Contains filtering rules for efficiency
 #   - Rules to drop prolific events of low-utility
 #   - Rules to restrict events based on `auid` constraints that would
 #     normally be applied to all rules
+# - `40_custom.rules`: Custom rules as defined by the ``auditd::custom_rules``
```
Comment (lnemsick-simp) [inline manifests/config/audit_profiles.pp]: I did not see `auditd::custom_rules` anywhere.  Looks like cruft from an earlier implementation?

File: manifests/init.pp (hunk 1 above)
Comment (trevor-vaughan) [reply manifests/init.pp]: Automatic alignment script snafu.

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-auditd/pipelines/59146643
